src/Paket/Program.fs

```diff
@@ -301,6 +301,83 @@ let push (results : ParseResults<_>) =
+            |> Seq.map(function 
+                | Paket.Requirements.AutoDetectFramework -> failwithf "couldn't detect framework"
+                | Paket.Requirements.FrameworkRestrictionList list ->
+                  list |> Seq.collect (
+                    function
+                    | Paket.Requirements.FrameworkRestriction.Exactly framework
+                    | Paket.Requirements.FrameworkRestriction.AtLeast framework -> Seq.singleton framework
+                    | Paket.Requirements.FrameworkRestriction.Between (bottom,top) -> [bottom; top] |> Seq.ofList //TODO: do we need to cap the list of generated frameworks based on this? also see todo in Requirements.fs for potential generation of range for 'between'
+                    | Paket.Requirements.FrameworkRestriction.Portable portable -> failwithf "unhandled portable framework %s" portable
+                  )
+              )
+            |> Seq.concat
+    )
+
+    let environmentFramework = lazy (
+        // HACK: resolve .net version based on environment
+        // list of match is incomplete / inaccurate
+        let version = Environment.Version
+        match version.Major, version.Minor, version.Build, version.Revision with
+        | 4, 0, 30319, 42000 -> DotNetFramework (FrameworkVersion.V4_6)
+        | 4, 0, 30319, _ -> DotNetFramework (FrameworkVersion.V4_5)
+        | _ -> DotNetFramework (FrameworkVersion.V4_5) // paket.exe is compiled for framework 4.5
+    )
+    let tupleMap f v = (v, f v)
+    let failOnMismatch toParse parsed f message =
+        if List.length toParse <> List.length parsed then
+            toParse
+            |> Seq.map (tupleMap f)
+            |> Seq.filter (snd >> Option.isNone)
+            |> Seq.map fst
+            |> String.concat ", "
+            |> sprintf "%s: %s. Cannot generate include scripts." message
+            |> failwith
+
+    let frameworksToGenerate =
+        let targetFrameworkList = providedFrameworks |> List.choose FrameworkDetection.Extract
+        
+        failOnMismatch providedFrameworks targetFrameworkList FrameworkDetection.Extract "Unrecognized Framework(s)"
+        
+        if targetFrameworkList |> Seq.isEmpty |> not then targetFrameworkList |> Seq.ofList
+        else if frameworksForDependencyGroups.Value |> Seq.isEmpty |> not then frameworksForDependencyGroups.Value 
+        else Seq.singleton environmentFramework.Value 
+    
+    let scriptTypesToGenerate = 
+      let parsedScriptTypes = providedScriptTypes |> List.choose Paket.LoadingScripts.ScriptGeneration.ScriptType.TryCreate
+      
+      failOnMismatch providedScriptTypes parsedScriptTypes Paket.LoadingScripts.ScriptGeneration.ScriptType.TryCreate "Unrecognized Script Type(s)"
+
+      match parsedScriptTypes with
+      | [] -> [Paket.LoadingScripts.ScriptGeneration.CSharp; Paket.LoadingScripts.ScriptGeneration.FSharp]
+      | xs -> xs
+
+    let workaround() = null |> ignore
```

Comment:
@robertpi had same question in https://github.com/fsharp/FSharp.Compiler.Service/pull/453